### ISSUE TITLE
Move React to peerDependencies (and maybe devDependencies)

### ISSUE THREAD

**Issue by Aaronius** [Lunuy/react-penpal#1]
Hey Lunuy, thanks again for setting up this project! I noticed that React is listed as a dependency in package.json. I would recommend moving it to `peerDependencies` and potentially `devDependencies` if you need it there as well for testing. The problem with having it in `dependencies` is that when a developer installs react-penpal as a dependency, react-penpal will bring React with it. It's likely the developer already has React as a dependency of their app, so they end up with two versions of React being loaded into their app, which causes errors. Thanks again! I added a reference to your project from Penpal's readme.

**Comment by Lunuy** [comment]
I didn't know that. Thanks for notifying.

FIXED - react-penpal@1.0.3

**Comment by Aaronius** [comment]
👍  Thanks! You may also want to consider expanding the React peer dependency semver range. As it is right now, if a developer installs react-penpal and their application is using a version of React that's lower than 17.0.1, they'll get an error (or maybe it's just a warning) because their version of React doesn't meet React-Penpal's React peer dependency range. The idea is that you want to make the semver range for the React peer dependency as wide as possible to support developers using a wide variety of React versions. In the case of React-Penpal, it looks like you need support for React hooks, so you might want to find out which version was the first version of React that started supporting hooks and then make your React peer dependency range include that version and anything afterward. Something like that. Sorry, I should have mentioned this previously.

**Comment by Aaronius** [comment]
I think @types/react also need to be moved from `dependencies` for the same reason you moved react from `dependencies`, but I really don't know what I'm talking about when it comes to TypeScript so I could very well be wrong.

**Comment by Lunuy** [comment]
```
  "devDependencies": {
    "react": "^17.0.1",
    ...
  },
  "dependencies": {
    "penpal": "^5.3.0"
  },
  "peerDependencies": {
    "react": "^16.8.0 || ^17.0.0"
  }
```
I referred to [react-hook-form's package.json](https://github.com/react-hook-form/react-hook-form/blob/master/package.json).

Fixed in 1.0.4
